### PR TITLE
Registered Scatter and Sum Op for SYCL

### PR DESCRIPTION
**Pull request by lukeiwanski**

### CODE REVIEW

File: tensorflow/core/kernels/reduction_ops_common.h [hunk 1]
```diff
@@ -66,7 +69,15 @@ struct Constants<CPUDevice> {
   const Eigen::IndexList<Eigen::type2index<1>> kOne;
   const Eigen::IndexList<Eigen::type2index<0>, Eigen::type2index<2>> kZeroTwo;
 };
-#endif
+#ifdef TENSORFLOW_USE_SYCL
```
Comment (benoitsteiner) [inline tensorflow/core/kernels/reduction_ops_common.h]: We should avoid duplicating this code. We should have one generic of struct Constants for the case where EIGEN_HAS_INDEX_LIST is defined and another for the case where it's not.

File: tensorflow/core/kernels/reduction_ops_common.h [hunk 2]
```diff
@@ -255,6 +266,24 @@ struct ReduceFunctor<CPUDevice, Reducer> {
   }
 };
 
+#ifdef TENSORFLOW_USE_SYCL
```
Comment (benoitsteiner) [inline tensorflow/core/kernels/reduction_ops_common.h]: This code should be shared with the cpu implementation

File: tensorflow/core/kernels/scatter_functor.h [hunk 3]
```diff
@@ -106,6 +109,32 @@ struct ScatterFunctor<CPUDevice, T, Index, op> {
   }
 };
 
+#ifdef TENSORFLOW_USE_SYCL
```
Comment (benoitsteiner) [inline tensorflow/core/kernels/scatter_functor.h]: Please share the code with the cpu implementation.

File: tensorflow/core/kernels/reduction_ops_common.h (hunk 1 above)
Comment (lukeiwanski) [inline tensorflow/core/kernels/reduction_ops_common.h]: FIxed in https://github.com/benoitsteiner/tensorflow-opencl/pull/25/commits/1f136132e6d03a5d47b9d736a5feecde1104bd0d

File: tensorflow/core/kernels/reduction_ops_common.h (hunk 2 above)
Comment (lukeiwanski) [inline tensorflow/core/kernels/reduction_ops_common.h]: Fixed in https://github.com/benoitsteiner/tensorflow-opencl/pull/25/commits/1f136132e6d03a5d47b9d736a5feecde1104bd0d

File: tensorflow/core/kernels/scatter_functor.h (hunk 3 above)
Comment (lukeiwanski) [inline tensorflow/core/kernels/scatter_functor.h]: Fixed in https://github.com/benoitsteiner/tensorflow-opencl/pull/25/commits/1f136132e6d03a5d47b9d736a5feecde1104bd0d

File: tensorflow/workspace.bzl [hunk 4]
```diff
@@ -17,11 +17,11 @@ def tf_workspace(path_prefix = "", tf_repo_name = ""):
   native.new_http_archive(
       name = "eigen_archive",
       urls = [
-          "http://bazel-mirror.storage.googleapis.com/bitbucket.org/eigen/eigen/get/feceaf0a7311.tar.gz",
-          "https://bitbucket.org/eigen/eigen/get/feceaf0a7311.tar.gz",
+          "http://bazel-mirror.storage.googleapis.com/bitbucket.org/benoitsteiner/opencl/get/59a1564642c8.tar.gz",
+          "https://bitbucket.org/benoitsteiner/opencl/get/59a1564642c8.tar.gz",
```
Comment (benoitsteiner) [inline tensorflow/workspace.bzl]: In this version the sycl test fail with a Fiji GPU. Can we stick to the current one at least for the time being ?

Comment (lukeiwanski) [inline tensorflow/workspace.bzl]: This should be fixed in: https://github.com/lukeiwanski/tensorflow-opencl/commit/6d0b5337f1c518846686d2ab329ac1ee1b661fe8

**Pull request merged**